### DISCUSSION TITLE
tud_cdc_connected()

### DISCUSSION THREAD

**Discussion by oznotes** [hathach/tinyusb#1173] [Q&A] [answered]
This works when I open the serial port with putty or with Arduino IDE but it does not work with QT serial port example ( terminus )

**Comment by mws-rmain** [comment, marked as answer]
tud_cdc_connected() reflects the state of 'DTR' (Data Terminal Ready) of the virtual UART. Check the QT code for a setting for 'hardware handshake', 'DTR/RTS', etc., and enable this. If there is no mention of it, see [QSerialPort Class](https://doc.qt.io/qt-5/qserialport.html), and look at setDataTerminalReady() (possibly the terminal example is not using DTR). Also setflowControl() can be used to set hardware flow control, but it looks like that only affects use of RTS/CTS.

**Reply by oznotes** [reply]
> tud_cdc_connected() reflects the state of 'DTR' (Data Terminal Ready) of the virtual UART. Check the QT code for a setting for 'hardware handshake', 'DTR/RTS', etc., and enable this. If there is no mention of it, see [QSerialPort Class](https://doc.qt.io/qt-5/qserialport.html), and look at setDataTerminalReady() (possibly the terminal example is not using DTR). Also setflowControl() can be used to set hardware flow control, but it looks like that only affects use of RTS/CTS.

setDataTerminalReady(true) fixes.
